connect/runtime/src/main/java/org/apache/kafka/connect/cli/ConnectDistributed.java

```diff
@@ -138,7 +141,7 @@ public Connect startConnect(Map<String, String> workerProps) {
         // herder is stopped. This is easier than having to track and own the lifecycle ourselves.
         DistributedHerder herder = new DistributedHerder(config, time, worker,
                 kafkaClusterId, statusBackingStore, configBackingStore,
-                advertisedUrl.toString(), connectorClientConfigOverridePolicy, sharedAdmin);
+                advertisedUrl.toString(), restClient, connectorClientConfigOverridePolicy, sharedAdmin, restClient);
```

Comment:
Isn't it a bit of an antipattern to have to pass in the client as part of the `uponShutdown` list? Is there any case where we'd want to instantiate a `DistributedHerder` with a client, but not have that client be shut down at the same time as the herder?

I'm wondering if we can just automatically close the client in `DistributedHerder::stop`.

Reply:
I was trying to follow the same pattern as the sharedAdmin, but I see that the situation isn't exactly analogous. The sharedAdmin isn't directly used by the DistributedHerder, while the restClient is.

Another way to think about it is that we are passing the object _without the responsibility to close it_ to the RestServer and the DistributedHerder, while we are explicitly passing the responsibility to close it as a lambda.

I'm fine with either implementation, and if you think it's more natural to explicitly close it, i'll make the change.

Reply:
I think it's more natural to close it in the herder, primarily because of the difference you've noted (where the shared admin client isn't used directly by the herder, but the REST client is).